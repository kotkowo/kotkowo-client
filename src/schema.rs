#[cynic::schema("kotkowo")]
mod schema {}

#[derive(cynic::QueryVariables, Debug)]
pub struct GetCatVariables<'a> {
    pub id: &'a cynic::Id,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "Query", variables = "GetCatVariables")]
pub struct GetCat {
    #[arguments(id: $id)]
    pub cat: Option<CatEntityResponse>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct CatEntityResponse {
    pub data: Option<CatEntity>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct CatEntity {
    pub attributes: Option<Cat>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct Cat {
    pub name: String,
}
