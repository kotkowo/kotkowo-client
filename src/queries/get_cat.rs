use crate::schema;

#[derive(cynic::QueryVariables, Debug)]
pub struct GetCatVariables<'a> {
    pub id: &'a cynic::Id,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "Query", variables = "GetCatVariables")]
pub struct GetCat {
    #[arguments(id: $id)]
    pub cat: Option<CatEntityResponse>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct CatEntityResponse {
    pub data: Option<CatEntity>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct CatEntity {
    pub attributes: Option<Cat>,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct Cat {
    pub name: String,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn cat_get_query() {
        use cynic::QueryBuilder;
        let id = &cynic::Id::new("test");
        let operation = GetCat::build(
            GetCatVariables{
                id,
        });
        insta::assert_snapshot!(operation.query);
    }
}
