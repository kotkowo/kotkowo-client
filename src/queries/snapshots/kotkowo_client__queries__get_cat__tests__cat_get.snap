---
source: src/queries/get_cat.rs
expression: operation.query
---
query GetCat($id: ID!) {
  cat(id: $id) {
    data {
      attributes {
        name
      }
    }
  }
}
